Sources/JSON.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  JSON.swift
+//  InstantSearchCore-iOS
+//
+//  Created by Vladislav Fitc on 04/03/2019.
+//  Copyright © 2019 Algolia. All rights reserved.
+//
+
+import Foundation
+
+public enum JSON: Equatable {
+    case string(String)
+    case number(Float)
+    case object([String:JSON])
```

Comment:
is it better to call that `dictionary` instead of `object`? WDYT?

Reply:
Agree